wsd/Storage.cpp

```diff
@@ -240,24 +241,10 @@ std::unique_ptr<StorageBase> StorageBase::create(const Poco::URI& uri, const std
             throw UnauthorizedRequestException("No acceptable WOPI hosts found matching the target host in config.");
         }
 #endif
-        if (FilesystemEnabled)
+        if (FilesystemEnabled || takeOwnership)
         {
-            return std::unique_ptr<StorageBase>(new LocalStorage(uri, jailRoot, jailPath));
-        }
-        else
-        {
-            // guard against attempts to escape
```

Comment:
Not possibly anymore. We flag this case via the "takeOwnership" flag, which is set when we get a convert-to request (via ConvertToBroker). So either the path is created by us (takeOwnership==true) or insecure filesystemEnabled is true. The URI doesn't come from the user in any case, and we don't need to detect convert-to cases (which could have been abused, hence the code below).

Reply:
I'd really be happier having a nicely encapsulated, opaque temp-file class (as you know) that makes this completely clear =) but for now if this fixes the leaks - and you're confident - sounds good.

Reply:
Agreed. For the filesystem case, for example, such a class will be needed indeed. But in practice filesystem isn't enabled (or common?), while for the convert-to and WOPI-like API we don't have temp files at all (as of this patch), so nothing to track.